### PR TITLE
chore: return inserted rows for each table in multi table insert

### PR DESCRIPTION
**Pull request by SkyFan2002**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

```sql
:) INSERT FIRST
    WHEN c3 = 5 THEN
      INTO t1
    WHEN c3 > 0 THEN
      INTO t2
SELECT * from s;

-[ RECORD 1 ]-----------------------------------
number of rows inserted into default.t1: 1
number of rows inserted into default.t2: 2
```

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

### CODE REVIEW

File: src/query/service/src/interpreters/interpreter_insert_multi_table.rs [hunk 1]
```diff
@@ -83,6 +88,18 @@ impl Interpreter for InsertMultiTableInterpreter {
...
         Ok(build_res)
     }
+
+    fn inject_result(&self) -> Result<SendableDataBlockStream> {
+        let mut columns = vec![];
+        let status = self.ctx.get_multi_table_insert_status();
+        let guard = status.lock();
+        for (tid, _) in &self.plan.target_tables {
+            let insert_rows = guard.insert_rows.get(tid).cloned().unwrap_or_default();
+            columns.push(UInt32Type::from_data(vec![insert_rows as u32]));
```
Comment (dantengsky) [inline src/query/service/src/interpreters/interpreter_insert_multi_table.rs:98]: should we use u64 instead?

File: src/query/sql/src/planner/plans/insert_multi_table.rs [hunk 2]
```diff
@@ -48,3 +53,17 @@ pub struct Into {
 pub struct Else {
     pub intos: Vec<Into>,
 }
+
+impl InsertMultiTable {
+    pub fn schema(&self) -> DataSchemaRef {
+        let mut fields = vec![];
+        for t in self.target_tables.iter() {
+            let field_name = format!("number of rows inserted into {}.{}", t.1.0, t.1.1);
```
Comment (dantengsky) [inline src/query/sql/src/planner/plans/insert_multi_table.rs:61]:
```suggestion
        for (_, (db, tbl)) in self.target_tables.iter() {
            let field_name = format!("number of rows inserted into {}.{}", db, tbl);
```

how about this?

**Review by youngsofun — APPROVED** [review]

File: src/query/sql/src/planner/binder/insert_multi_table.rs [hunk 3]
```diff
@@ -125,26 +129,35 @@ impl Binder {
...
+                into_clauses,
+                source_schema.clone(),
+                &mut source_bind_context,
+                &mut target_tables,
+            )
+            .await?;
+
+        let mut ordered_target_tables = target_tables.into_iter().collect::<Vec<_>>();
+        // convenient for testing
+        ordered_target_tables.sort_by_key(|(_, (d, t))| format!("{}.{}", d, t));
```
Comment (dantengsky) [inline src/query/sql/src/planner/binder/insert_multi_table.rs:151]:
```suggestion
        ordered_target_tables.sort_by(|(_, l), (_, r)| l.0.cmp(&r.0).then_with(|| l.1.cmp(&r.1)));
```

how about this?

File: src/query/sql/src/planner/plans/insert_multi_table.rs (hunk 2 above)
Comment (SkyFan2002) [reply src/query/sql/src/planner/plans/insert_multi_table.rs:61]: Thanks

**Review by dantengsky — APPROVED** [review]

**Pull request merged**